artifacts/writer.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+"""The artifact writer objects."""
+
+import abc
+import json
+import yaml
+
+class BaseArtifactsWriter(object):
+  """Class that implements the artifacts reader interface."""
+
+  @abc.abstractmethod
+  def _ConvertArtifact(self, artifact):
+    """Converts artifacts to dict from ArtifactDefinition objects
+
+    Args:
+      artifact: an instances of ArtifactDefinition
+
+    Yields:
+      dict representation of ArtifactDefinition instance
+    """
+
+  @abc.abstractmethod
+  def WriteArtifacts(self, artifacts, filename):
+    """Reads artifact definitions from a file.
+
+    Args:
+      artifacts: list of ArtifactDefinition objects to be written
+
+      filename: filename to write artifacts to
+
+    Yields:
+      writes artifacts to file
+    """
+
+  @abc.abstractmethod
+  def FormatArtifacts(self, artifacts):
+    """Formats artifact desired output format
```

Comment:
Internally yes.  The Artificer server only uses FormatArtifacts to prepare YAML to serve rather than writing a file anywhere.